lib/src/supabase_auth.dart

```diff
@@ -350,6 +355,43 @@ extension GoTrueClientSignInProvider on GoTrueClient {
       return result;
     }
   }
+
+  /// Signs a user in using native Apple Login.
+  ///
+  /// This method only works on iOS. If you want to sign in a user using Apple
+  /// on other platforms, please use the `signInWithOAuth` method.
+  ///
+  /// This method is experimental as the underlying `signInWithIdToken` method is experimental.
+  @experimental
+  Future<AuthResponse> signInWithApple() async {
+    assert(Platform.isIOS, 'Please use signInWithOAuth for non-iOS platforms');
```

Comment:
```suggestion
    assert(!kIsWeb && Platform.isIOS, 'Please use signInWithOAuth for non-iOS platforms');
```
I think otherwise the assert itself would throw and I don't know if the message would come through.